### PR TITLE
[SFSRAP-20] Added test for mappings between package properties and th…

### PR DESCRIPTION
**Pull request by HSAR**
…e plugin manifest

### CODE REVIEW

File: RoslynPluginGenerator/AnalyzerPluginGenerator.cs [hunk 1]
```diff
@@ -195,11 +195,6 @@ private static PluginManifest CreatePluginDefinition(IPackage package)
             pluginDefn.Organization = GetValidManifestString(ListToString(package.Owners));
             pluginDefn.Version = GetValidManifestString(package.Version.ToNormalizedString());
 
-            //pluginDefn.IssueTrackerUrl
-            //pluginDefn.License;
-            //pluginDefn.SourcesUrl;
-            //pluginDefn.TermsConditionsUrl;
```
Comment (HSAR) [inline RoslynPluginGenerator/AnalyzerPluginGenerator.cs]: Removed these reminders of unused properties.

File: Tests/IntegrationTests/Roslyn/RoslynGenTests.cs [hunk 2]
```diff
@@ -36,7 +37,7 @@ public void RoslynPlugin_GenerateForValidAnalyzer_Succeeds()
             string packageId = "analyzer1.pkgid1";
             string localNuGetDir = TestUtils.CreateTestDirectory(this.TestContext, ".localNuGet");
             IPackageManager localNuGetStore = CreatePackageManager(localNuGetDir);
-            AddPackage(localNuGetStore, packageId, "1.0.2", analyzer.GetType().Assembly.Location);
+            IPackage analyzerPkg =  AddPackage(localNuGetStore, packageId, "1.0.2", analyzer.GetType().Assembly.Location);
```
Comment (HSAR) [inline Tests/IntegrationTests/Roslyn/RoslynGenTests.cs]: Need to return a reference to the package so we can check the plugin manifest against it.

File: Tests/IntegrationTests/Roslyn/RoslynGenTests.cs [hunk 3]
```diff
@@ -90,27 +94,33 @@ private IPackageManager CreatePackageManager(string rootDir)
...
+        private IPackage AddPackage(IPackageManager manager, string id, string version, string payloadAssemblyFilePath)
         {
             PackageBuilder builder = new PackageBuilder();
             builder.Id = id;
+            builder.Title = "dummy title";
             builder.Version = new SemanticVersion(version);
             builder.Description = "dummy description";
             builder.Authors.Add("dummy author");
+            builder.Owners.Add("dummy owner");
+            builder.ProjectUrl = new System.Uri("http://dummyurl/");
```
Comment (HSAR) [inline Tests/IntegrationTests/Roslyn/RoslynGenTests.cs]: Additional properties so that we can verify correct mappings.

**Comment by HSAR** [comment]
There is a note on the JIRA ticket to give an example of these mappings, but the README.md is the only place for documentation at this time and it doesn't obviously have a place to go within said README.

File: Tests/IntegrationTests/Roslyn/RoslynGenTests.cs [hunk 4]
```diff
@@ -90,27 +94,33 @@ private IPackageManager CreatePackageManager(string rootDir)
...
 
             PhysicalPackageFile file = new PhysicalPackageFile();
             file.SourcePath = payloadAssemblyFilePath;
             file.TargetPath = "analyzers/" + Path.GetFileName(payloadAssemblyFilePath);
             builder.Files.Add(file);
 
+            ZipPackage pkg;
```
Comment (bgavrilMS) [inline Tests/IntegrationTests/Roslyn/RoslynGenTests.cs]: You don't need to define pkg outside the using block, you can return after InstallPackage. the stream is guaranteed to be disposed.

File: Tests/IntegrationTests/Roslyn/RoslynGenTests.cs [hunk 5]
```diff
@@ -90,27 +94,33 @@ private IPackageManager CreatePackageManager(string rootDir)
...
-        private void AddPackage(IPackageManager manager, string id, string version, string payloadAssemblyFilePath)
+        private IPackage AddPackage(IPackageManager manager, string id, string version, string payloadAssemblyFilePath)
         {
             PackageBuilder builder = new PackageBuilder();
             builder.Id = id;
+            builder.Title = "dummy title";
             builder.Version = new SemanticVersion(version);
             builder.Description = "dummy description";
             builder.Authors.Add("dummy author");
+            builder.Owners.Add("dummy owner");
```
Comment (bgavrilMS) [inline Tests/IntegrationTests/Roslyn/RoslynGenTests.cs]: Since you only have one test, it would be better to have several owners listed.

**Comment by bgavrilMS** [comment]
LGTM

**Comment by duncanpMS** [comment]
Documentation: just add a section to the readme, or add a new file if you prefer.

**Pull request merged**